control-plane/agents/src/bin/core/volume/specs.rs

```diff
@@ -828,6 +829,53 @@ impl ResourceSpecsLocked {
         }
     }
 
+    pub(crate) fn check_capacity_limit_for_resize(
```

Comment:
How is this enforcing the limit when we have >1 concurrent calls?

Reply:
It's taking exclusive `write` lock on all specs just like get_or_create_volume does, while checking limits. Anything more needed here?

Reply:
That's just while checking it, but you could have 2 request come in a the same time, they each take turns checking, they both think they're in the resize usage, but the last one goes over the limit

Reply:
hmm yeah, for that it seems to me that we need to seal the gap between this check and acquiring volumespec operation guard for resize. That can be achieved by acquiring volumespec operation guard from `check_capacity_limit_for_resize` before the RAII guard on specs gets dropped, if we are within capacity limit, and return that guard from this function. wdyt?

Reply:
This PR now contains only reconciler changes.